### PR TITLE
improve error messages for mmrm failure in jackknife or bayes methods

### LINKED ISSUE #237: Nicer error message when MMRM fails in Bayes
When `method <- method_bayes()`, if MMRM fails, th error message that appears is:

`Error in stan_data$R %*% mmrm_initial$beta : requires numeric/complex matrix/vector arguments`

Indeed, `mmrm_initial` is a list of one element: `mmrm_initial$failed` which is `TRUE`.

A nicer error message, like the one we use for analogous case for the `condmean(type = "bootstrap")` methods, would be better:

`Fitting MMRM to original dataset failed`

Also `condmean(type = "jackknife")` does not distinguish between error in MMRM fit in the original dataset or in a leave-one-out sample: the error message is the same and is the following: `Jackknife sample failed`

### PR DESCRIPTION
**Pull request by nociale**
Closes #237.

### CODE REVIEW

**Comment by nociale** [comment]
Last commit also addresses #238

File: R/draws.R [hunk 1]
```diff
@@ -329,7 +327,9 @@ get_jackknife_draws <- function(longdata, method) {
             optimizer = optimizer
         )
         if (sample$failed) {
-            stop("Jackknife sample failed")
+            stop(sprintf("Fitting MMRM to jackknife sample without subject with ID %s failed",
```
Comment (gowerc) [inline R/draws.R:330]:
```suggestion
            stop(sprintf("MMRM failed to fit to the data after removing subject %s",
```

**Comment by nociale** [comment]
@gowerc Can we merge?

File: R/draws.R [hunk 2]
```diff
@@ -325,7 +327,9 @@ get_jackknife_draws <- function(longdata, method) {
             optimizer = optimizer
         )
         if (sample$failed) {
-            stop("Jackknife sample failed")
+            stop(sprintf("MMRM failed to fit to the data after removing subject %s",
+                         ids[i])
```
Comment (gowerc) [inline R/draws.R:331]:
```suggestion
            stop(
                sprintf("MMRM failed to fit to the data after removing subject %s", ids[i])
```

File: R/mcmc.R [hunk 3]
```diff
@@ -62,14 +54,16 @@ fit_mcmc <- function(
     group,
     subjid,
     visit,
-    same_cov,
-    n_imputations,
-    burn_in,
-    burn_between,
-    verbose = TRUE,
-    seed = NULL
+    method
```
Comment (gowerc) [inline R/mcmc.R:57]: I'm suprised this didn't require any changes to our unit tests. Do we not have anything for fit_mcmc ?

File: R/mcmc.R [hunk 4]
```diff
@@ -83,6 +77,10 @@ fit_mcmc <- function(
         optimizer = c("L-BFGS-B", "BFGS")
     )
 
+    if (mmrm_initial$failed) {
+        stop("Fitting MMRM to original dataset failed")
+    }
+
```
Comment (gowerc) [inline R/mcmc.R:83]: What happened if it failed before this ?  Would it just carry on processing ?

File: R/mcmc.R (hunk 3 above)
Comment (nociale) [reply R/mcmc.R:57]: I just checked that we have some tests for fit_mcmc on nightly mode.. I will make changes accordingly

File: R/mcmc.R (hunk 4 above)
Comment (nociale) [reply R/mcmc.R:83]: It was giving a less interpretable error when operating with the regression coefficients which where NA probably: 

`Error in stan_data$R %*% mmrm_initial$beta : requires numeric/complex matrix/vector arguments`

File: R/mcmc.R (hunk 3 above)
Comment (nociale) [reply R/mcmc.R:57]: ok, just pushed the changes

File: R/mcmc.R (hunk 4 above)
Comment (gowerc) [reply R/mcmc.R:83]: ouch , yer good to have this here now !

Could you include a test for a model which can't converge and show that this catches it though please.  I know its trivial but its good to have encase we edit the code in the future and break this for some reason

Comment (nociale) [reply R/mcmc.R:83]: Ok, I will do it!

Comment (nociale) [reply R/mcmc.R:83]: Done!

**Review by gowerc — APPROVED** [review]

**Pull request merged**